analyzers/tests/SonarAnalyzer.Test/TestCases/AvoidUnderPosting.CSharp12.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+using System.ComponentModel.DataAnnotations;
+
+public class ClassNotUsedInRequests
+{
+    int ValueProperty { get; set; }                                             // Compliant
+}
+
+public struct Struct { public int Foo { get; set; } }
+public record struct RecordStruct { public int Foo { get; set; } }
+
+public class ModelUsedInController
```

Comment:
You also need a Model hierachy, becasue the ITypeSymbol only contains the member of the current class not from base classes.
```
public class ModelBase
{
// .. Some properties with violations
}
public class Model1 : ModelBase
{
// .. Some violations
}
public class Model2 : ModelBase
{
// .. Some violations
}

[HttpGet] void Action1(ModelBase model) { }
[HttpGet] void Action2(Model1 model) { }
[HttpGet] void Action3(Model2 model) { }
```